### PR TITLE
Be robust to HTTP errors

### PR DESCRIPTION
**Pull request by zerebubuth**
Adds a configuration parameter for the number of retries to attempt when HTTP connections fail or return one of the whitelisted error status codes.

Connects to #60.

@nvkelso could you review, please?

### CODE REVIEW

**Comment by nvkelso** [comment]
Over to Aaron.

File: tilequeue/command.py [hunk 1]
```diff
@@ -940,6 +940,7 @@ def tilequeue_process_wof_neighbourhoods(cfg, peripherals):
         wof_cfg['macrohoods-meta-url'],
         wof_cfg['data-prefix-url'],
         n_raw_neighbourhood_fetch_threads,
+        wof_cfg.get('max-retries', 0)
```
Comment (nvkelso) [inline tilequeue/command.py]: Talking with MattA, requires coordination with Chef repo config (TBD).

File: tilequeue/wof.py [hunk 2]
```diff
@@ -101,8 +101,35 @@ def parse_neighbourhood_meta_csv(csv_line_generator, placetype):
...
 
 
-def fetch_wof_url_meta_neighbourhoods(url, placetype):
-    r = requests.get(url, stream=True)
+def _make_requests_session_with_retries(max_retries):
+    from requests.adapters import HTTPAdapter
+    from requests.packages.urllib3.util import Retry
+
+    s = requests.Session()
+    a = HTTPAdapter(
```
Comment (nvkelso) [inline tilequeue/wof.py]: Should we wait a (small) period of time before retrying?

Comment (zerebubuth) [inline tilequeue/wof.py]: Good idea - added in 90c6f37.

**Comment by nvkelso** [comment]
LGTM

**Comment by nvkelso** [comment]
(Please also link the Chef changes here.)

**Comment by zerebubuth** [comment]
Chef changes are in [this PR](https://github.com/mapzen/chef-tilequeue/pull/18), and will be enabled on dev.

**Pull request merged**